android-gif-drawable/src/main/java/pl/droidsonroids/gif/GifDrawable.kt

```diff
@@ -0,0 +1,983 @@
+package pl.droidsonroids.gif
+
+import android.content.ContentResolver
+import android.content.res.AssetFileDescriptor
+import android.content.res.AssetManager
+import android.content.res.ColorStateList
+import android.content.res.Resources
+import android.content.res.Resources.NotFoundException
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.ColorFilter
+import android.graphics.Paint
+import android.graphics.PixelFormat
+import android.graphics.PorterDuff
+import android.graphics.PorterDuffColorFilter
+import android.graphics.Rect
+import android.graphics.drawable.Animatable
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import android.os.Build
+import android.os.SystemClock
+import android.widget.MediaController.MediaPlayerControl
+import androidx.annotation.DrawableRes
+import androidx.annotation.FloatRange
+import androidx.annotation.IntRange
+import androidx.annotation.RawRes
+import pl.droidsonroids.gif.GifDrawable
+import pl.droidsonroids.gif.GifError.Companion.fromCode
+import pl.droidsonroids.gif.GifInfoHandle.Companion.openUri
+import pl.droidsonroids.gif.GifViewUtils.getDensityScale
+import pl.droidsonroids.gif.transforms.CornerRadiusTransform
+import pl.droidsonroids.gif.transforms.Transform
+import java.io.File
+import java.io.FileDescriptor
+import java.io.IOException
+import java.io.InputStream
+import java.nio.ByteBuffer
+import java.util.Locale
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.ScheduledThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+import kotlin.jvm.Throws
+import kotlin.math.max
+
+/**
+ * A [Drawable] which can be used to hold GIF images, especially animations.
+ * Basic GIF metadata can also be examined.
+ *
+ * @author koral--
+ */
+class GifDrawable internal constructor(
+    val mNativeInfoHandle: GifInfoHandle,
+    oldDrawable: GifDrawable?,
+    executor: ScheduledThreadPoolExecutor?,
+    val mIsRenderingTriggeredOnDraw: Boolean
+) : Drawable(), Animatable, MediaPlayerControl {
+
+    val mExecutor: ScheduledThreadPoolExecutor
+
+    @Volatile
+    var mIsRunning = true
+    var mNextFrameRenderTime = Long.MIN_VALUE
+    private val mDstRect = Rect()
+
+    /**
+     * Paint used to draw on a Canvas
+     */
+    private val paint = Paint(Paint.FILTER_BITMAP_FLAG or Paint.DITHER_FLAG)
+
+    /**
+     * Frame buffer, holds current frame.
+     */
+    var mBuffer: Bitmap
```

Comment:
It should be internal. This also applies to most (if not all) other members which were previously package private.
But, it may break the backwards compatibility.
Please hold on any changes in this matter for now. I need to think about that.